xcode/Subconscious/Shared/Models/Petname.swift

```diff
@@ -63,66 +45,172 @@ public struct Petname:
+        self.parts = [name]
     }
     
     /// Join a list of petnames into a dotted string, i.e. [foo, bar, baz] -> foo.bar.baz
     /// Names are joined in order of their appearance in `petnames`
-    public init(petnames: [Petname]) {
-        let petnamePath = petnames
-            .map({ s in s.verbatim })
-            .joined(separator: ".")
-        self.description = petnamePath.lowercased()
-        self.verbatim = petnamePath
+    public init?(parts: [Petname.Name]) {
+        guard !parts.isEmpty else {
+            return nil
+        }
+        
+        guard !parts.contains(where: { part in part.verbatim.isEmpty }) else {
+            return nil
+        }
+        
+        self.parts = parts
     }
     
-    /// Convert a string into a petname.
-    /// This will sanitize the string as best it can to create a valid petname.
-    public init?(formatting string: String) {
-        self.init(Self.format(string))
+    public init?(_ description: String) {
+        let parts = description.components(separatedBy: Self.separator)
+        let mappedParts = parts.compactMap { part in Petname.Name(part) }
+        guard parts.count == mappedParts.count else {
+            return nil
+        }
+        
+        self.init(parts: mappedParts)
     }
     
-    /// Explode a petname path into the individual steps along the way, in written order.
-    /// i.e. `@foo.bar.baz` -> `[foo, bar, baz]`
-    public func parts() -> [Petname] {
-        verbatim
-            .split(separator: ".")
-            .compactMap { part in Petname(part.toString()) }
-    }
     
-    /// Return a new petname with a numerical suffix.
-    /// A plain petname e.g. `ziggy` becomes `ziggy-1`
-    /// But `ziggy-1` becomes `ziggy-2` etc.
-    public func increment() -> Petname? {
-        guard let match = description.wholeMatch(of: Self.numberedSuffixRegex),
-              let separator = match.output.separator else {
-            return Petname(formatting: verbatim + "-1")
+    /// Convert a string into a petname.
+    /// This will sanitize the string as best it can to create a valid petname.
+    public init?(formatting string: String) {
+        let parts = string.split(separator: Self.separator)
+        let mappedParts = parts.compactMap { part in Petname.Name(formatting: part) }
+        guard parts.count == mappedParts.count else {
+            return nil
         }
         
-        if let numberString = match.output.suffix,
-           let number = Int(numberString) {
-            return Petname(formatting: "\(match.output.petname)\(separator)\(String(number + 1))")
-        } else {
-            return Petname(formatting: "\(match.output.petname)\(separator)1")
-        }
+        self.init(parts: mappedParts)
+    }
+     
+    /// Combines two petnames to build up a traversal path
+    /// i.e. `Petname("foo")!.append(petname: Petname.Part("bar")!)` -> `bar.foo`
+    public func append(name: Petname.Name) -> Petname? {
```

Comment:
Because of the confusing ordering (append prepends), I might suggest another verb. Ideas:

- join `alice.join(bob)`
- subjoin (join under)